### PR TITLE
CA-293335: compat code cleanups, remove layering violation

### PR DESCRIPTION
**Pull request by gaborigloi**

### CODE REVIEW

**Comment by gaborigloi** [comment]
I've tested attach / detach / sr create in xencenter and they all worked :smile:

File: main.ml [hunk 1]
```diff
@@ -194,13 +196,27 @@ end = struct
...
+      end
+      else id
     in
     return (Ok (device_config, compat_in, compat_out))
+
+  let sr_attach ~uuid device_config =
+    compat_uri device_config >>>= fun compat_in ->
+    let device_config =
+      if !V.version = Some pvs_version then
+        ("sr_uuid", uuid) :: device_config
```
Comment (edwintorok) [inline main.ml]: PVS did not use `sr_uuid`, that was something that older versions of our own xapi-storage-plugins used.
I think you can safely drop `sr_uuid` from here, and keep just `compat_uri`

**Comment by gaborigloi** [comment]
Thanks, I've edited my first commit.
I've tested reattach of a forgotten SR, and it was complaining about missing sr_uuid, hopefully I just had a too old xencenter that did not have the fix.

**Comment by gaborigloi** [comment]
Played around with the fixed XenCenter, now everything works :)

**Pull request merged**